chef/cookbooks/cpe_screensaver/resources/cpe_screensaver.rb

```diff
@@ -39,30 +46,181 @@
     'PayloadOrganization' => organization,
     'PayloadVersion' => 1,
     'PayloadDisplayName' => 'Screensaver',
-    'PayloadContent' => [
-      {
-        'PayloadType' => 'com.apple.ManagedClient.preferences',
-        'PayloadVersion' => 1,
-        'PayloadIdentifier' => "#{prefix}.screensaver",
-        'PayloadUUID' => '3B2AD6A9-F99E-4813-980A-4147617B2E75',
-        'PayloadEnabled' => true,
-        'PayloadDisplayName' => 'ScreenSaver',
-        'PayloadContent' => {
-          'com.apple.screensaver' => {
-            'Forced' => [
-              {
-                'mcx_preference_settings' => {
+    'PayloadContent' => [],
+  }
+
+  # idleTime can be any arbitrary value in this PayloadContent as this hash key
+  # simply locks the UI element, the real value is set in the ByHost hash, i'm
+  # setting it here to match node['cpe_screensaver']['idleTime'] for consistency
+  screensaver_profile['PayloadContent'].push(
```

Comment:
There has to be a cleaner way to do this. Would this create multiple profiles? If so, I think this is the wrong approach.

On Aug 29, 2016, at 8:36 AM, Nate Walck <notifications@github.com<mailto:notifications@github.com>> wrote:


In chef/cookbooks/cpe_screensaver/resources/cpe_screensaver.rb<https://github.com/facebook/IT-CPE/pull/47#discussion_r76629101>:

> -        'PayloadIdentifier' => "#{prefix}.screensaver",
> -        'PayloadUUID' => '3B2AD6A9-F99E-4813-980A-4147617B2E75',
> -        'PayloadEnabled' => true,
> -        'PayloadDisplayName' => 'ScreenSaver',
> -        'PayloadContent' => {
> -          'com.apple.screensaver' => {
> -            'Forced' => [
> -              {
> -                'mcx_preference_settings' => {
> +    'PayloadContent' => [],
> +  }
> +
> +  # idleTime can be any arbitrary value in this PayloadContent as this hash key
> +  # simply locks the UI element, the real value is set in the ByHost hash, i'm
> +  # setting it here to match node['cpe_screensaver']['idleTime'] for consistency
> +  screensaver_profile['PayloadContent'].push(


We prefer << instead of .push for appending arrays.

-
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/facebook/IT-CPE/pull/47/files/e178288abe91de78d24326c3dc43f38a2efa9719#r76629101>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFa_GGckqUzya9nZEmh49T7G9415suLzks5qkvxwgaJpZM4JvNqo>.